marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Safety.hs

```diff
@@ -233,7 +235,9 @@ checkTransaction solveConstraints version@MarloweV1 marloweContext@MarloweContex
       let
         walletContext = walletForConstraints version marloweContext changeAddress constraints
       pure
-        . either (pure . TransactionValidationError transaction . show) (const $ TransactionWarning <$> V1.txOutWarnings txOutput)
+        . either
+          (pure . TransactionValidationError transaction . show)
+          (const $ TransactionWarning transaction <$> V1.txOutWarnings txOutput)
```

Comment:
Failures in the analysis of safety are server errors, but the results of successful safety analysis are returned as `[SafetyError]`.